sale_purchase_history/models/purchase_order.py

```diff
@@ -0,0 +1,19 @@
+from odoo import fields, models, api, _
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    def action_view_history(self):
+        return {
+            'name': _('View History'),
+            'res_model': 'product.sale.history',
+            'view_mode': 'form',
+            'view_id': self.env.ref('sale_purchase_history.product_sale_history_form').id,
+            # 'res_id':
+            'context': {
+                'active_model': 'purchase.order',
+            },
```

Comment:
I think you don't need this anymore since the context is passed from the button.
Also, active_model and active_id are used in the standard framework so don't use them unless you intend to play with the standard behavior